ledger/src/blockstore_db.rs

```diff
@@ -1227,6 +1245,39 @@ impl TypedColumn for columns::OptimisticSlots {
     type Type = blockstore_meta::OptimisticSlotMetaVersioned;
 }
 
+impl Column for columns::MerkleRootMeta {
+    type Index = (Slot, /*fec_set_index:*/ u32);
+
+    fn index(key: &[u8]) -> Self::Index {
+        let slot = BigEndian::read_u64(&key[..8]);
+        let fec_set_index = BigEndian::read_u32(&key[8..]);
+
+        (slot, fec_set_index)
+    }
+
+    fn key((slot, fec_set_index): Self::Index) -> Vec<u8> {
+        let mut key = vec![0; 16];
```

Comment:
This is not 16 bytes anymore.
It is 8 + 4 = 12 bytes.

Reply:
good catch

Reply:
Can we add a test for this part?  The mismatch isn't caught by any test sounds scary.

Reply:
unable to create a test that would fail. 
`assert_eq!(MerkleRootMeta::index(&MerkleRootMeta::key((slot, fec_set_index)), (slot, fec_set_index))` will pass with both `12` and `16`. 

only way would be to check the length of `MerkleRootMeta::key` which requires hardcoding the `16` or `12` in the unit test. this approach would not have caught the oversight.

Reply:
a round-trip test might still be useful to assert that what is written is read back the same.
asserting manually in the test that the key size is 12 is also fine (agreed that it wouldn't have caught this though, but nonetheless)

Reply:
Never hit submit but was chatting with Ashwin and agreed with others that a unit test would have been unlikely to catch this.

There is a `key_size()` field on the `Column` trait that I thought about suggesting; however, this method is flawed in that it calls `std::mem::size_of` on `Self::Index`. The result is that the calculated key size from this call will include any alignment bytes that would be present in `Self::Index`. But, we manually pack the keys end-to-end, so the actual keys do not contain alignment bytes. Have a PR to rip that out / maybe re-implement on each column individually.

Will think on this one a little more to see if there is some way to make this a little more full-proof.